### PR TITLE
HOSTEDCP-501: Added detailed documentation about Hypershift release process

### PR DESCRIPTION
**Pull request by jparrill**
This PR contains the documentation related with Release process of Hypershift and CPO. It does not involves only Hypershift repo but openshift/release one, which means that the consequence will involve, new jobs in the CI, new payloads for external teams, changes in the images generated and some modifications in the release branches

**Which issue(s) this PR fixes**
Fixes #[HOSTEDCP-501](https://issues.redhat.com/browse/HOSTEDCP-501)

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.
- [x] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

**Comment by enxebre** [comment]
looking great, thanks!

File: docs/content/contribute/release-process.md [hunk 1]
```diff
@@ -4,6 +4,15 @@ title: Release process
 
 # Release process
 
+!!! important
+
+    This is a complex process that involve some changes in multiple repositories and will affect multiple teams daily basis work.
```
Comment (bryan-cox) [inline docs/content/contribute/release-process.md:9]:
```suggestion
    This is a complex process that involves some changes in multiple repositories and will affect multiple teams daily basis work.
```

File: docs/content/contribute/release-process.md [hunk 2]
```diff
@@ -19,17 +28,103 @@ The HO has an independent release cadence. For consumer products:
   - Generate release notes:
     - `GITHUB_ACCESS_TOKEN=XXXX FROM=v0.1.0 TO=v0.1.1 make release`
     - Use the output to create the PR for bump the new image in the product gitOps repo. E.g.
+
+This is a sample of how looks like the release notes added to the PR:
```
Comment (bryan-cox) [inline docs/content/contribute/release-process.md:32]:
```suggestion
This is a sample of how the release notes look when added to the PR:
```

File: docs/content/contribute/release-process.md [hunk 3]
```diff
@@ -19,17 +28,103 @@ The HO has an independent release cadence. For consumer products:
...
   - [Validate Network Input for HostedCluster](https://github.com/openshift/hypershift/pull/2215)
-  ```
\ No newline at end of file
+  ```
+
+### Supported Versions
+
+This change involve basically the OCP supported versions by Hypershift operator:
+
+- `support/supportedversion/version.go` which contain the variable called `LatestSupportedVersion`. This one contains as you can imagine, the Latest supported version. We need to put the new version` here.
```
Comment (bryan-cox) [inline docs/content/contribute/release-process.md:53]:
```suggestion
- `support/supportedversion/version.go` which contains the variable called `LatestSupportedVersion`. This one contains, as you can imagine, the Latest supported version. We need to put the new version here.
```

File: docs/content/contribute/release-process.md [hunk 4]
```diff
@@ -19,17 +28,103 @@ The HO has an independent release cadence. For consumer products:
...
+- `mce/kubevirt/destroy/hypershift-mce-kubevirt-destroy-chain.yaml`
+- `mce/aws/destroy/hypershift-mce-aws-destroy-chain.yaml`
+- `mce/aws/create/hypershift-mce-aws-create-chain.yaml`
+- `agent/create/hostedcluster/hypershift-agent-create-hostedcluster-ref.yaml`
+- `agent/destroy/hypershift-agent-destroy-chain.yaml`
+- `ibmcloud/install/hypershift-ibmcloud-install-chain.yaml`
+
+### Mirroring Images for CPO and HO
+
+For folder `core-services/image-mirroring/hypershift/` we need to add a mapping file for each version we want to mirror. This is a sample about how looks like this kind of [file](https://github.com/openshift/release/blob/master/core-services/image-mirroring/hypershift/mapping_hypershift_4_13). This sample points to version `4.13` and the filename is `mapping_hypershift_4_13`:
```
Comment (bryan-cox) [inline docs/content/contribute/release-process.md:87]:
```suggestion
For folder `core-services/image-mirroring/hypershift/` we need to add a mapping file for each version we want to mirror. This is a sample of how it looks like this kind of [file](https://github.com/openshift/release/blob/master/core-services/image-mirroring/hypershift/mapping_hypershift_4_13). This sample points to version `4.13` and the filename is `mapping_hypershift_4_13`:
```

File: docs/content/contribute/release-process.md [hunk 5]
```diff
@@ -19,17 +28,103 @@ The HO has an independent release cadence. For consumer products:
...
+registry.ci.openshift.org/ocp-arm64/4.13:hypershift quay.io/hypershift/hypershift:4.13-arm64 quay.io/hypershift-arm64/hypershift:latest-arm64
+registry.ci.openshift.org/ocp-arm64/4.13:hypershift-operator quay.io/hypershift/hypershift-operator:4.13-arm64
+```
+
+As you can see, the version is hardcoded but, what happens to the `latest` version?:
+
+```
+registry.ci.openshift.org/ocp/4.14:hypershift quay.io/hypershift/hypershift:4.14 quay.io/hypershift/hypershift:latest
+registry.ci.openshift.org/ocp/4.14:hypershift-operator quay.io/hypershift/hypershift-operator:4.14  quay.io/hypershift/hypershift-operator:latest
+registry.ci.openshift.org/ocp-arm64/4.14:hypershift quay.io/hypershift/hypershift:4.14-arm64 quay.io/hypershift-arm64/hypershift:latest-arm64
```
Comment (bryan-cox) [inline docs/content/contribute/release-process.md:101]:
```suggestion
registry.ci.openshift.org/ocp-arm64/4.14:hypershift quay.io/hypershift/hypershift:4.14-arm64 quay.io/hypershift/hypershift:latest-arm64
```
That repo was a typo and should be resolved once https://github.com/openshift/release/pull/36933 is merged.

Comment (jparrill) [reply docs/content/contribute/release-process.md:101]: Cool, nice catch

File: docs/content/contribute/release-process.md [hunk 6]
```diff
@@ -19,17 +28,103 @@ The HO has an independent release cadence. For consumer products:
...
   - [Validate Network Input for HostedCluster](https://github.com/openshift/hypershift/pull/2215)
-  ```
\ No newline at end of file
+  ```
+
+### Supported Versions
+
+This change involves basically the OCP supported versions by Hypershift operator:
+
+- `support/supportedversion/version.go` which contains the variable called `LatestSupportedVersion`. This one contains, as you can imagine, the Latest supported version. We need to put the new version` here.
```
Comment (bryan-cox) [inline docs/content/contribute/release-process.md:53]:
```suggestion
- `support/supportedversion/version.go` which contains the variable called `LatestSupportedVersion`. This one contains, as you can imagine, the Latest supported version. We need to put the new version here.
```

**Pull request merged**